### PR TITLE
Added field type check to sanitise middleware

### PR DESCRIPTION
**Pull request by MCrawleyHomeOffice**
Added sanitisation middleware to strip out unwanted/malicious data from user input.

### CODE REVIEW

File: controller/base-controller.js [hunk 1]
```diff
@@ -162,6 +164,25 @@ module.exports = class BaseController extends EventEmitter {
     return validator(key, req.form.values[key], req.form.values, emptyValue);
   }
 
+  _sanitize(req, res, callback) {
+    // If we don't have any data, no need to progress
+    if(req.form.values) {
+      Object.keys(req.form.values).forEach(function (property, propertyIndex) {
+        // If it's not a string, don't sanitise it
+        if(_.isString(req.form.values[property])) {
```
Comment (Alex-Swann) [inline controller/base-controller.js:172]: might be good to have some unit tests to check for values passed to _sanitize in the req.form.values that aren't strings so if this was removed those tests would break

File: controller/base-controller.js [hunk 2]
```diff
@@ -162,6 +164,25 @@ module.exports = class BaseController extends EventEmitter {
     return validator(key, req.form.values[key], req.form.values, emptyValue);
   }
 
+  _sanitize(req, res, callback) {
+    // If we don't have any data, no need to progress
+    if(req.form.values) {
```
Comment (Alex-Swann) [inline controller/base-controller.js:169]: Also a test to check for empty req.form.values in case this is removed too. This is always an array right?

Comment (MCrawleyHomeOffice) [reply controller/base-controller.js:169]: It's an object not an array so I will check for an empty object (ie. {}) as well as undefined.

File: controller/base-controller.js (hunk 1 above)
Comment (MCrawleyHomeOffice) [reply controller/base-controller.js:172]: Good shout, done.

File: controller/base-controller.js (hunk 2 above)
Comment (Alex-Swann) [reply controller/base-controller.js:169]: Maybe:
```
if(!_.isEmpty(req.form.values)) {
```

Comment (Alex-Swann) [reply controller/base-controller.js:169]: Then add a test for `req.form.values = {}` and make sure the unit tests do as they should.

**Review by Rhodine — APPROVED** [review]
lg
